contracts/contracts/coordination/BqETHSubscription.sol

```diff
@@ -0,0 +1,224 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./Coordinator.sol";
+import "./IFeeModel.sol";
+
+/**
+ * @title BqETH Subscription
+ * @notice Manages the subscription information for rituals.
+ */
+contract BqETHSubscription is IFeeModel {
+    using SafeERC20 for IERC20;
+
+    Coordinator public immutable coordinator;
+    IERC20 public immutable feeToken;
+
+    uint32 public constant INACTIVE_RITUAL_ID = type(uint32).max;
+
+    // TODO: DAO Treasury
+    // TODO: Should it be updatable?
+    address public immutable beneficiary;
```

Comment:
I'd rename this to `treasury`. Beneficiary is a loaded term from the staking contract.

Reply:
In one of the next PR I've changed it to Ownable contract to be able to transfer this role, and I'll rename rest to treasury in that PR 👍